### PR TITLE
Move sippy under /apps

### PR DESCRIPTION
**Pull request by jimdaga**
## Resolves:
* https://github.com/kubernetes/k8s.io/issues/2150 (first PR of a few)

## Changes:
* Move `sippy` directory under the new `/apps` path. 
* Move yaml out of the secondary `sippy` dir, into the root `sippy` dir (aka: `apps/sippy` vs `apps/sippy/sippy`)
* Add `app:` label to resources that didn't already have it. 
* Re-order content under `metadata:` to have a consistent order 

## Notes:
* Checked https://cs.k8s.io/ and didn't find any references to these files, so no external dependencies require updates. 
* This "app" looks to be an easy starting point while I get feedback from others. I plan to move the other "apps" per the ISSUE request, but will handle that in multiple smaller PRs.

### CODE REVIEW

**Comment by ameukam** [comment]
@jimdaga Thank you so much for doing this!

Can you move those YAML files to `apps/sippy` instead of `apps/sippy/sippy` ?

**Comment by jimdaga** [comment]
> @jimdaga Thank you so much for doing this!
> 
> Can you move those YAML files to `apps/sippy` instead of `apps/sippy/sippy` ?

@ameukam no problem, happy to be involved.

This app was actually the reason I asked in #wg-k8s-infra about the idea of storing yaml under a `config` dir. I agree that having `apps/sippy/sippy` feels odd (I left the folder layout the way it was originally). But if just moving the yaml to the `apps/sippy` root is fine, I can make that change.

**Comment by ameukam** [comment]
> > @jimdaga Thank you so much for doing this!
> > Can you move those YAML files to `apps/sippy` instead of `apps/sippy/sippy` ?
> 
> @ameukam no problem, happy to be involved.
> 
> This app was actually the reason I asked in #wg-k8s-infra about the idea of storing yaml under a `config` dir. I agree that having `apps/sippy/sippy` feels odd (I left the folder layout the way it was originally). But if just moving the yaml to the `apps/sippy` root is fine, I can make that change.

I think you can proceed. Once the WIP is removed, we'll leave the approval to the OWNERS. 
cc @spiffxp

**Comment by ameukam** [comment]
/ok-to-test
/lgtm
/hold 

/assign @wojtek-t @deads2k 
cc @stevekuznetsov

File: apps/sippy/deployment.yaml
Comment (spiffxp) [inline apps/sippy/deployment.yaml:7]: I think this ordering makes more sense to a human, but it will mean automated enforcement of yaml formatting/ordering can't be done with jq/yq.

I see the type is ordered this way in https://git.k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L116-L288 so presumably we could pass through `kubectl --dry-run` before something that enforced spacing/indenting standards

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm
I may `/hold cancel` to move this forward once I'm done watching over some other deploys.

We ought to spend some time fleshing out what the standard layout should be in https://github.com/kubernetes/k8s.io/issues/2150, though I'm fine iterating our way there with a few apps at first

File: apps/sippy/deployment.yaml
Comment (jimdaga) [reply apps/sippy/deployment.yaml:7]: Fair point. I'll admit I usually base things on how `helm create` generates their default yaml, but I'm totally fine with reverting the order if it helps our automation. 

```
$ cat testchart/templates/deployment.yaml | head -6
apiVersion: apps/v1
kind: Deployment
metadata:
  name: {{ include "testchart.fullname" . }}
  labels:
    {{- include "testchart.labels" . | nindent 4 }}
```

Let me know if I should re-order, and I can get another commit added today. 

Thanks!

**Pull request merged**
